dsp-shared/src/main/scala/dsp/valueobjects/Id.scala

```diff
@@ -13,6 +13,54 @@ import java.util.UUID
 sealed trait Id
 object Id {
 
+  /**
+   * Stores the role's ID.
+   *
+   * @param uuid the role's UUID
+   * @param iri the role's IRI
+   */
+  sealed abstract case class RoleId private (
+    uuid: UUID,
+    iri: Iri.RoleIri
+  ) extends Id
+
+  object RoleId {
+    private val roleIriPrefix = "http://rdfh.ch/roles/"
+
+    /**
+     * Generates a RoleId instance with a UUID and a given IRI.
+     *
+     * @param iri the role's IRI used to create RoleId
+     * @return new RoleId instance
+     */
+    def fromIri(iri: Iri.RoleIri): Validation[Throwable, RoleId] = {
+      val uuid: UUID = UUID.fromString(iri.value.split("/").last)
```

Comment:
in my PR I wrapped this in a Try because in theory these methods could throw... but I'm not sure they ever will, so maybe it's also overkill.

Reply:
Ok